src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -163,38 +62,63 @@ function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: strin
   )
 }
 
-function useActiveWalletProvider(): Web3Provider | undefined {
-  return getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+interface ActiveWeb3ProviderProps {
+  provider?: Eip1193Provider | JsonRpcProvider
+  jsonRpcEndpoint?: string | JsonRpcProvider
+}
+
+export function ActiveWeb3Provider({
+  provider: propsProvider,
+  jsonRpcEndpoint,
+  children,
+}: PropsWithChildren<ActiveWeb3ProviderProps>) {
+  const metaMaskConnection = useMemo(
+    () => toWeb3Connection(initializeConnector<MetaMask>((actions) => new MetaMask({ actions }))),
+    []
+  )
+  const walletConnectConnectionQR = useMemo(() => getWalletConnectConnection(false, jsonRpcEndpoint), [jsonRpcEndpoint]) // WC via tile QR code scan
```

Comment:
why do you need different connectors for these two?

Reply:
being able to make our own QR code tile requires disabling the `qrcode` option during WalletConnect instantiation

<img width="362" alt="Screen Shot 2022-07-01 at 4 33 02 PM" src="https://user-images.githubusercontent.com/27475332/176964535-1512d410-827e-49d2-a7f1-54e64d11dcfe.png">
<img width="366" alt="Screen Shot 2022-07-01 at 4 33 10 PM" src="https://user-images.githubusercontent.com/27475332/176964546-2a046bcc-c927-4980-8d86-b595a7495e2f.png">

Reply:
wow that's kinda brutal